Fix realm import failure when OID4VCI credential-offer-create role exists

Awambeng opened the pull request:
This PR modifies setupRealmDefaults() to check for credential-offer-create role in the realm representation before creating it during import. This prevents ModelDuplicateException when importing a realm that already contains the built-in OID4VCI role.

The fix follows the same pattern used for offline_access role handling:
- Added overloaded setupRealmDefaults() method that accepts optional RealmRepresentation parameter
- Check hasRealmRole() before creating credential-offer-create role
- Updated importRealm() to pass realm representation to setupRealmDefaults()
- Preserves backward compatibility for fresh realm creation

Added testRealmImportWithOID4VCICredentialOfferCreateRole() to verify that realm import succeeds when the role is present in exported JSON.

Closes #44637

Awambeng committed:
Fix realm import failure when OID4VCI credential-offer-create role exists

Modify setupRealmDefaults() to check for credential-offer-create role
in the realm representation before creating it during import. This
prevents ModelDuplicateException when importing a realm that already
contains the built-in OID4VCI role.

The fix follows the same pattern used for offline_access role handling:
- Added overloaded setupRealmDefaults() method that accepts optional
  RealmRepresentation parameter
- Check hasRealmRole() before creating credential-offer-create role
- Updated importRealm() to pass realm representation to setupRealmDefaults()
- Preserves backward compatibility for fresh realm creation

Added testRealmImportWithOID4VCICredentialOfferCreateRole() to verify
that realm import succeeds when the role is present in exported JSON.

Closes #44637

Signed-off-by: Awambeng Rodrick <awambengrodrick@gmail.com>
Signed-off-by: Awambeng <awambengrodrick@gmail.com>

Fix import violations

Signed-off-by: Awambeng <awambengrodrick@gmail.com>

Captain-P-Goldfish commented on testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/exportimport/ExportImportTest.java:
```diff
@@ -320,6 +324,62 @@ public void testExportUserProfileConfig() throws IOException {
+        exportImport.runExport();
+        
+        // Verify the exported file contains the role
+        Map<String, RealmRepresentation> exportedRealms = ImportUtils.getRealmsFromStream(JsonSerialization.mapper, new FileInputStream(targetFilePath));
```
The FileInputStream is not getting closed by the method `ImportUtils.getRealmsFromStream`. You need to close it manually.

Awambeng committed:
Refactor ExportImportTest to use try-with-resources for FileInputStream

- Updated the assertExportContainsGoogleClientSecret and export verification methods to utilize try-with-resources for FileInputStream. This change ensures proper resource management and enhances code readability.

Signed-off-by: Awambeng Rodrick <awambengrodrick@gmail.com>

Captain-P-Goldfish reviewed (approved):
LGTM
